src/core/qgsproject.h

```diff
@@ -951,6 +975,8 @@ class CORE_EXPORT QgsProject : public QObject, public QgsExpressionContextGenera
 
     QMap<QString, QgsMapLayer *> mMapLayers;
 
+    QgsWeakMapLayerPointerList mLayerOrder;
```

Comment:
Does it really makes any difference? Happy to, but @wonder-sk preferred it here.